### ISSUE TITLE
Warmup Schduler

### ISSUE THREAD

**Issue by ken19980727** [s920128/NAR-BERT-ASR#3]
Seems the optimizer learning rate doesn't matter with the warmup schduler

```
class WarmupLR(_LRScheduler):
    def __init__(
        self,
        optimizer: torch.optim.Optimizer,
        warmup_steps: Union[int, float] = 25000,
        last_epoch: int = -1,
    ):
        self.warmup_steps = warmup_steps
        super().__init__(optimizer, last_epoch)

    def __repr__(self):
        return f"{self.__class__.__name__}(warmup_steps={self.warmup_steps})"

    def get_lr(self):
        step_num = self.last_epoch + 1
        return [
            (768) ** -0.5
            * min(step_num ** -0.5, step_num * self.warmup_steps ** -1.5)
            for lr in self.base_lrs
        ]
```
Am I right?

**Comment by s920128** [comment]
optimizer  learning rate  is given by scheduler.
```
optimizer =torch.optim.Adam(model.parameters(), lr=0.001)
scheduler = WarmupLR(optimizer, 12000)
```
When training process, you can find that the `scheduler.step()` is followed after `optimizer.step()`
```
optimizer.step()
scheduler.step()
```

**Comment by ken19980727** [comment]
> optimizer learning rate is given by scheduler.
> 
> ```
> optimizer =torch.optim.Adam(model.parameters(), lr=0.001)
> scheduler = WarmupLR(optimizer, 12000)
> ```
> 
> When training process, you can find that the `scheduler.step()` is followed after `optimizer.step()`
> 
> ```
> optimizer.step()
> scheduler.step()
> ```

Thanks for your response @s920128 !
I knew it.
But I'm curious about the 0.001
No matter how much I change it, it will not affect the learning rate curve.

**Comment by s920128** [comment]
It just for convenience! (given a small  learning rate as initialize value)
Any small value **may** not affect the learning rate curve. Because  learning rate will be given by scheduler immediately.

**Comment by ken19980727** [comment]
Got it.
Thanks for your reply!
